lib/planet.rb

```diff
@@ -0,0 +1,30 @@
+class Planet
+  attr_reader :name, :color, :mass_kg, :distance_from_sun_km, :fun_fact
+
+  def initialize(name, color, mass_kg, distance_from_sun_km, fun_fact)
+    @name = name
+    @color = color
+    @mass_kg = number_input_check(mass_kg.to_s,
+      "ERROR: Mass must be a number greater than zero.
+      Enter mass in kg: ")
+    @distance_from_sun_km = number_input_check(distance_from_sun_km.to_s,
+      "ERROR: Distance must be a number greater than zero.
+      Enter distance from sun in km: ")
+    @fun_fact = fun_fact
+  end
+
+  def number_input_check(input, error_msg)
+    while ((/\A[+-]?\d*(\.?\d+)?(e?\+?\d+)?\Z/.match(input) ==
+      nil) || input.to_f <= 0)
+      print error_msg
+      input = gets.chomp
```

Comment:
This should instead of doing a `gets.chomp` raise an exception and let `main.rb` deal with the user.